src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackSerializer.NonGeneric.cs

```diff
@@ -237,12 +238,14 @@ internal CompiledMethods(Type type)
                     MessagePackWriterSerialize lambda = Expression.Lambda<MessagePackWriterSerialize>(body, param1, param2, param3).Compile(PreferInterpretation);
 
                     this.Serialize_MessagePackWriter_T_Options = lambda;
+                    this.Serialize_MessagePackWriter_T_Options = (MessagePackWriterSerialize)serialize.CreateDelegate(typeof(MessagePackWriterSerialize));
 #endif
                 }
 
                 {
                     // public static T Deserialize<T>(ref MessagePackReader reader, MessagePackSerializerOptions options)
                     MethodInfo deserialize = GetMethod(nameof(Deserialize), type, new Type[] { typeof(MessagePackReader).MakeByRefType(), typeof(MessagePackSerializerOptions) });
+                    MethodInfo deserialize = GetMethod(nameof(DeserializeSemiGeneric), type, new Type[] { typeof(MessagePackReader).MakeByRefType(), typeof(MessagePackSerializerOptions) });
```

Comment:
Again, you're declaring a local for the second time. Compile error.

Reply:
Using AOT for ios is not working. As I wrote in [1344](https://github.com/neuecc/MessagePack-CSharp/issues/1344), the release crashes. After researching the issue, I realized that in order for everything to work it is necessary to enable "enable mono interpreter". This option is enabled by default in debug mode.

Reply:
> Using AOT for ios is not working. As I wrote in https://github.com/neuecc/MessagePack-CSharp/issues/1344, the release crashes.

I can't find any comment from you that tells about how AOT is not working. I only see you describing a runtime failure. Do you know about AOT? Using the mpc or the msbuild package to pre-generate code so that at runtime no code generation is required?
I suggest we continue that discussion over at that issue.

Reply:
Line 247 need to be deleted.